src/AppInstallerRepositoryCore/RepositorySource.cpp

```diff
@@ -49,6 +49,12 @@ namespace AppInstaller::Repository
             return std::find_if(sources.begin(), sources.end(), [&name](const SourceDetailsInternal& sd) { return Utility::ICUCaseInsensitiveEquals(sd.Name, name); });
         }
 
+        // Finds a source from the given vector by its name. Returned source cannot be modified.
+        auto FindSourceByNameConst(const std::vector<SourceDetailsInternal>& sources, std::string_view name)

```

Comment:
>FindSourceByNameConst [](start = 13, length = 21)

There shouldn't be a need to make a new function; just make the one above take a const vector.

Reply:
I tried initially to add const to input vector. Then I found existing code used FindSourceByName() and then modifys the source details pointed by returned itr. So I created this const version one.

---
In reply to: [518347549](https://github.com/microsoft/winget-cli/pull/634#discussion_r518347549) [](ancestors = 518347549)